docs/proposals/exec-logs.md

```diff
@@ -0,0 +1,141 @@
+	+ 1-2 Re-establish a websocket tunnel between cloudcore and edgecore and supports TLS authentication.
+		
+	This proposal chooses the second solution. The new tunnel is specifically responsible for the forwarding of streaming data, which can be used as `/metric` data transmission tunnel without affecting the logic of cloudhub and edgehub. 
+
+	In most cases, we consider users to use `exec and logs` commands for debugging purposes. In edge computing scenarios, disconnection due to network instability is considered acceptable. Therefore, in the first stage, we do not support the reliable transmission of messages in this tunnel. 
+	
+	For cloudcore, we need to add a tunnel module, which listens on port 10350 by default (different from 10250), and needs to support websocket and spdy protocols. Both websocket connection requests from edgecore and apiserver connection requests must be supported.
+	
+	For edgecore, we need to add a tunnel module to initiate a websocket connection to the cloud tunnel. After receiving the data from the cloud, the tunnel calls the CRI interface to communicate with the container runtime.
+
+The process is as follows:
+	<img src="../images/proposals/kubeedge-exec-con3.png">
+
+
+For the data transmission in the tunnel proxy, there are several considerations:
+
+- In order to distinguish the data of `logs, exec, /metric` connection types, we need to add type definition to distinguish these data.
+- Distinguish which connection the data belongs to. We need to use a session-like definition.
+- cloudcore tunnel needs to support SPDY and websocket protocols.
+- Supports TLS authentication.
+
+The message structure is as follows (working in process ...)：
+
+```
+type TunnelMsg struct {
+	ID          int64
+	ConnID      int64
+	messageType int 
+	data		[]byte
+}
+```
+
+
+2.	Redirect requests made by kube-apiserver to the cloud tunnel service
+
+	We will not modify the code of kube-apiserver, so the connection request address initiated by kube-apiserver is {edgenodeip}:10250, and the 10250 port is getted by (`node.Status.DaemonEndpoints.KubeletEndpoint.Port`), so we need DNAT.
+
+```
+ kubectl get node {nodename} -o=jsonpath="{.status.daemonEndpoints.kubeletEndpoint['Port']}"
+```
+
+    First of all, we need to modify the logic of edged module. When updating the node status, set the `Status.DaemonEndpoints.KubeletEndpoint.Port` to 10350 by default, so for all edge nodes, kube-apiserver will request 10350 port, which can distinguish the connection of cloud node (10250 port).
+
+On all hosts where kube-apiserver is running, we need to execute the following command:
+
+```shell
+
+iptables -t nat -A OUTPUT -p tcp  --dport 10350 -j DNAT --to {cloudcoreip}:10350
+
+```
+
+The flowchart is as follows:
+<img src="../images/proposals/kubeedge-exec-con4.png">
+
+
+## Detail Design
+
+working in process ...
```

Comment:
@daixiang0 can you update this as well?

Reply:
I do not have premission to wirte this pr :(